tracer/src/Datadog.Trace/Configuration/ImmutableTracerSettings.cs

```diff
@@ -70,9 +69,14 @@ public ImmutableTracerSettings(TracerSettings settings)
             // create dictionary copy without "env" or "version",
             // these value are used for "Environment" and "ServiceVersion"
             // or overriden with DD_ENV and DD_VERSION
-            var globalTags = settings.GlobalTags
-                                     .Where(kvp => kvp.Key is not (Tags.Env or Tags.Version))
-                                     .ToDictionary(kvp => kvp.Key, kvp => kvp.Value);
+            var globalTags = new Dictionary<string, string>();
```

Comment:
Can we initialize this `Dictionary<>` with the correct size from `settings.GlobalTags`?

Reply:
done in [9b271d6](https://github.com/DataDog/dd-trace-dotnet/pull/3735/commits/9b271d6330489f7f37df13159a5f41abe6ef8bb8)